components/tidb_query/src/rpn_expr/impl_math.rs

```diff
@@ -463,6 +463,46 @@ pub fn round_dec(arg: &Option<Decimal>) -> Result<Option<Decimal>> {
     }
 }
 
+#[inline]
+#[rpn_fn]
+pub fn truncate_real_with_int(arg0: &Option<Real>, arg1: &Option<Int>) -> Result<Option<Real>> {
+    match (arg0, arg1) {
+        (&Some(x), &Some(d)) => {
+            let d = if d >= 0 {
+                d.min(i64::from(i32::max_value())) as i32
```

Comment:
Could you please explain why we need to compare `d` with the bounds of `i32`? In `TiDB`, it just converts `i64` into `i32` with overflow.